scripts/fixtures.sql

```diff
@@ -0,0 +1,28 @@
+create procedure department(name text)
+language plpgsql
+as $$
+declare
+    root realms.id%type;
+    y_root realms.id%type;
+begin
+    insert into realms (name, parent)
+    values (name, 0)
+    returning id into root;
+
+    for y in 2020 .. 2021 loop
+        insert into realms (name, parent)
+        values (y, root)
+        returning id into y_root;
+
+        insert into realms (name, parent)
+        values ('Summer', y_root);
+        insert into realms (name, parent)
+        values ('Winter', y_root);
+    end loop;
+end; $$;
+
+call department('Mathematics');
+call department('Computer Science');
+call department('Physics');
+
+drop procedure department;
```

Comment:
Mhhh, tbh, I'm not sure if this procedure is worth the code duplication it avoids. Also, this is temporary code, code duplication whatever. So I would much more prefer something like this (pseudo code):

```
var root = select ...;
var maths = insert('mathematics', root);
var cs = ...;
...
insert('2021', maths);
...
```

In the above code there would still be an `insert` helper procedure. But all test data is easily editable. And it's easy to add more stuff.

Furthermore... under "root", there should be "lectures", "events", ... and stuff like that; not lectures.

Reply:
So, I started with a simpler version as you suggested and promptly made a mistake. So I tend to disagree with this; it's not just avoiding duplication because it's nicer in theory, but because it literally already helped to avoid mistakes. I think extending it is still simple enough.

And Re:

> Furthermore... under "root", there should be "lectures", "events", ... and stuff like that; not lectures.

What does "should" even mean here? If that's always how it has to be, we can hard code it and don't need it in the realm tree. Maybe my university doesn't have events?

In all seriousness, though, I can of course add another layer for you if you want it, but you should not rely on that being there in any way, I think.

Reply:
Mh, then I will task you later with adding lots of other dummy data ;-) I actually would like to have a pretty large amount of dummy data not just a handful of courses or so. I think have closer to real life data helps with development.

Regardind "should": yep, that was the wrong word to use here. I just meant to say that most institutions would probably do it that way. Of course, everyone can do whatever with the realm tree.

Reply:
Sure, bring it. ;D

/me is a PL/pgSQL guru now :sunglasses: